uint8 channel
float64 degrees